packages/react-instantsearch-dom/src/core/createInstantSearchServer.js

```diff
@@ -116,7 +115,7 @@ const createInstantSearchServer = algoliasearch => {
 
     const helper = algoliasearchHelper(client, sharedParameters.index);
 
-    if (isEmpty(derivedParameters)) {
+    if (Object.keys(derivedParameters).length === 0) {
```

Comment:
We can simply the condition to `!Object.keys(derivedParameters).length` the return value is always a number.